### PR TITLE
[FLINK-31507][table] Move execution logic of ShowOperation out from TableEnvironmentImpl

### PR DESCRIPTION
**Pull request by wuchong**
## What is the purpose of the change

The PR continues the work started at https://github.com/apache/flink/pull/22175

This PR does this for `ShowOperation`s and `DescribeTableOperation`.

## Brief change log

- Migrate `ShowOperation` to extend `ExecutableOperation` and move it's execution logic out from `TableEnvironmentImpl`.
- `DescribeTableOperation` is  a special case of `ShowColumnsOperation`, so we migrate it in this PR as well.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.
## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-table/flink-sql-gateway/src/test/resources/sql/table.q [hunk 1]
```diff
@@ -31,12 +31,12 @@ org.apache.flink.table.api.ValidationException: Table with identifier 'default_c
...
 describe non_exist;
 !output
-org.apache.flink.table.api.ValidationException: Tables or views with the identifier 'default_catalog.default_database.non_exist' doesn't exist
+org.apache.flink.table.api.ValidationException: Tables or views with the identifier 'default_catalog.default_database.non_exist' doesn't exist.
 !error
 
 desc non_exist;
 !output
-org.apache.flink.table.api.ValidationException: Tables or views with the identifier 'default_catalog.default_database.non_exist' doesn't exist
+org.apache.flink.table.api.ValidationException: Tables or views with the identifier 'default_catalog.default_database.non_exist' doesn't exist.
```
Comment (snuyanzin) [inline flink-table/flink-sql-gateway/src/test/resources/sql/table.q:39]: it looks like same should be applied for sql client's version

Comment (chucheng92) [reply flink-table/flink-sql-gateway/src/test/resources/sql/table.q:39]: +1 flink-sql-client-test need to add this either. then CliClientITCase will call to run it.

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ExecutableOperation.java [hunk 2]
```diff
@@ -57,5 +59,9 @@ interface Context {
         FunctionCatalog getFunctionCatalog();
 
         ModuleManager getModuleManager();
+
+        ResourceManager getResourceManager();
```
Comment (chucheng92) [inline flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ExecutableOperation.java:63]: it looks like we can get catalog and module or jars. how about showJobs ?

File: flink-table/flink-sql-gateway/src/test/resources/sql/table.q (hunk 1 above)
Comment (wuchong) [reply flink-table/flink-sql-gateway/src/test/resources/sql/table.q:39]: Yes, this is missed. I just fixed it and rebased it to master.

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ExecutableOperation.java (hunk 2 above)
Comment (wuchong) [reply flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ExecutableOperation.java]: Currently, `ShowJobsOperation` is implemented in SQL Gateway, not in `TableEnvironemntImpl`. And it relies on the additional `JobClient` to fetch jobs that require more effort. So I would like to migrate it in another issue.

**Comment by wuchong** [comment]
Hi @snuyanzin , @chucheng92 , I have fixed the tests. Do you have other concerns?

**Comment by snuyanzin** [comment]
@chucheng92 may be I missed something, however I wonder if it's not sorted here https://github.com/apache/flink/pull/22215/files#diff-745d3739f888f4a240798978a54933ff1a7d8d48b96a1e9472c7a9007c9f523aR132-R142

**Comment by chucheng92** [comment]
@snuyanzin pls see master: https://github.com/apache/flink/blob/master/flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java  listTables().

**Comment by wuchong** [comment]
Yes, @chucheng92 , the new implementation also sorts the table list (in the useLike block).

**Comment by chucheng92** [comment]
> Yes, @chucheng92 , the new implementation also sorts the table list (in the useLike block).

Sorry, found that in the useLike block.  looks good from my side

**Review by chucheng92 — APPROVED** [review]
LGTM

**Comment by wuchong** [comment]
Merging...

**Pull request closed without merging**